samples/snippets/csharp/VS_Snippets_CLR/etwtracelarge/cs/program.cs

```diff
@@ -0,0 +1,202 @@
+//<root>
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+
+namespace Demo
+{
+    //<InterfaceSource>
+    public interface IMyLogging
+    {
+        void Error(int errorCode, string message);
+        void Warning(string message);
+    }
+
+    public sealed class MySource : EventSource, IMyLogging
+    {
+        public static MySource Log = new();
+
+        [Event(1)]
+        public void Error(int errorCode, string message) => WriteEvent(1, errorCode, message);
+
+        [Event(2)]
+        public void Warning(string message) => WriteEvent(2, message);
+    }
+    //</InterfaceSource>
+
+    //<UtilitySource>
+    public abstract class UtilBaseEventSource : EventSource
+    {
+        protected UtilBaseEventSource() 
+            : base()
+        { }
+
+        protected UtilBaseEventSource(bool throwOnEventWriteErrors)
+            : base(throwOnEventWriteErrors)
+        { }
+
+        // helper overload of WriteEvent for optimizing writing an event containing
+        // payload properties that don't align with a provided overload. This prevents
+        // EventSource from using the object[] overload which is expensive.
+        protected unsafe void WriteEvent(int eventId, int arg1, short arg2, long arg3)
+        {
+            if (IsEnabled())
+            {
+                EventSource.EventData* descrs = stackalloc EventSource.EventData[3];
+                descrs[0].DataPointer = (IntPtr)(&arg1);
+                descrs[0].Size = 4;
```

Comment:
Would be good to set EventData.Reserved = 0 for each of these.  Normally, these are zero'd, but for those that write code in assemblies that don't clear locals, this can break ETW.

Reply:
The `EventData.Reserved` property is marked `internal` so users implementing an `EventSource` can't access it.

The comment on the property says:
```
            /// <summary>
            /// Reserved by ETW.  This property is present to ensure that we can zero it
            /// since System.Private.CoreLib uses are not zero'd.
            /// </summary>
```

which is true for all use cases inside `EventSource`.

https://github.com/dotnet/runtime/blob/36b6247cc7743dda997e9d671175ddc2b3e030ba/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs#L1202-L1259

Based on this, we should probably change that to have a default initializer to 0 or make the property public.

Reply:
Ah, that's a good point - it is internal.  We should have a default initializer.  Not an issue for this PR then.

Reply:
I'll spin a PR over on the runtime to add that